packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx

```diff
@@ -0,0 +1,119 @@
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useLayoutEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ScrollView, StyleSheet } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import i18n from 'src/i18n'
+import { addressToDisplayNameSelector } from 'src/identity/selectors'
+import { HeaderTitleWithSubtitle } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { rewardsSendersSelector } from 'src/recipients/reducer'
+import useSelector from 'src/redux/useSelector'
+import { tokensByCurrencySelector } from 'src/tokens/selectors'
+import TransferSentContent from 'src/transactions/feed/detailContent/TransferSentContent'
+import {
+  TokenExchange,
+  TokenTransaction,
+  TokenTransactionTypeV2,
+  TokenTransfer,
+} from 'src/transactions/types'
+import { Currency } from 'src/utils/currencies'
+import { getDatetimeDisplayString } from 'src/utils/time'
+import RewardReceivedContent from './detailContent/RewardReceivedContent'
+import TransferReceivedContent from './detailContent/TransferReceivedContent'
+
+type Props = StackScreenProps<StackParamList, Screens.TransactionDetailsScreen>
+
+function useHeaderTitle(transaction: TokenTransaction) {
+  const { t } = useTranslation()
+  const tokensByCurrency = useSelector(tokensByCurrencySelector)
+  const celoAddress = tokensByCurrency[Currency.Celo]?.address
+  const rewardsSenders = useSelector(rewardsSendersSelector)
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+
+  switch (transaction.type) {
+    case TokenTransactionTypeV2.Exchange:
+      // TODO: Change this to show any exchanges, not just CELO sell/purchase.
+      const isCeloSell = (transaction as TokenExchange).inAmount.tokenAddress === celoAddress
+      return isCeloSell ? i18n.t('soldGold') : i18n.t('purchasedGold')
+    case TokenTransactionTypeV2.Sent:
+      const isCeloSend = (transaction as TokenTransfer).amount.tokenAddress === celoAddress
+      return t(isCeloSend ? 'transactionHeaderWithdrewCelo' : 'transactionHeaderSent')
```

Comment:
We should prefer static `t()` expressions to more easily migrate to a tool extracting strings from the codebase.
```suggestion
      return isCeloSend ? t('transactionHeaderWithdrewCelo') : t('transactionHeaderSent')
```